evalml/tests/automl_tests/test_iterative_algorithm.py

```diff
@@ -30,6 +31,16 @@ def test_iterative_algorithm_allowed_pipelines(logistic_regression_binary_pipeli
     assert algo.allowed_pipelines == allowed_pipelines
 
 
+class MockEstimator(Estimator):
```

Comment:
We have a fixture for a mock classifier called `dummy_classifier_estimator_class` would that work in this tests?